example/variable_raycast_collisions/data/player.script

```diff
@@ -0,0 +1,51 @@
+local platypus = require"platypus.platypus"
+
+function init(self)
+  msg.post(".", "acquire_input_focus")
+  self.input_state = {}
+  self.platypus = platypus.create({
+      collisions = {
+        separation = platypus.SEPARATION_RAYS,
+        groups = {
+          [hash("ground")] = platypus.DIR_ALL,
+        },
+        left = 4, right = 4, top = 7, bottom = 6, offset = vmath.vector3(0, -1, 0)
+      },
+      gravity = -800,
+      max_velocity = 300,
+      allow_double_jump = true,
+      allow_wall_jump = true,
+      allow_wall_slide = true,
+      wall_slide_gravity = -50,
+      debug = true,
+    })
+  self.current_animation = nil
+
+end
+
+function update(self, dt)
+	self.platypus.update(dt)
+end
+
+function on_message(self, message_id, message, sender)
+  self.platypus.on_message(message_id, message)
+end
+
+function change_sprite(self, anim, collisions)
+  msg.post("#player", "play_animation", { id=hash(anim) })
+  self.platypus.define_collisions(collisions)
+end
+
+function on_input(self, action_id, action)
+  local move_table = {
```

Comment:
This should for performance reasons be moved outside the on_input() function

Reply:
it was for readability, but ok for performances